dev/com.ibm.ws.security.wim.adapter.ldap/src/com/ibm/ws/security/wim/adapter/ldap/LdapConfigManager.java

```diff
@@ -1031,20 +1040,39 @@ private void setGroupMemberFilter() {
 
     }
 
-    private void setLoginProperties(String loginProps) {
+    private void setLoginProperties(Map<String, Object> configProps) throws WIMSystemException {
         if (iPersonAccountTypes.size() > 0) {
             LdapEntity acct = getLdapEntity(iPersonAccountTypes.get(iPersonAccountTypes.size() - 1));
             iLoginAttrs = new ArrayList<String>();
             iLoginProps = new ArrayList<String>();
-            if (loginProps != null) {
-                String[] loginProperties = loginProps.split(";");
-                for (int i = 0; i < loginProperties.length; i++) {
-                    String propName = loginProperties[i];
+            List<Map<String, Object>> loginProps = Nester.nest(ConfigConstants.CONFIG_PROP_LOGIN_PROPERTIES, configProps);
+
+            if (loginProps != null && !loginProps.isEmpty()) {
+                isLoginPropertyDefined = true;
+                List<String> unsupportedProps = new ArrayList<String>();
+                for (Map<String, Object> propDO : loginProps) {
+                    String propName = (String) propDO.get(ConfigConstants.CONFIG_PROP_NAME);
+                    // Check if the property is supported or not
+                    List<String> props = new ArrayList<String>(1);
+                    props.add(propName);
+                    List<String> supportedProp = getSupportedProperties(acct, props);
+                    if (supportedProp == null || supportedProp.size() == 0) {
+                        unsupportedProps.add(propName);
+                    }
                     iLoginAttrs.add(getAttributeName(acct, propName));
```

Comment:
Don't add it if it is not supported. I realize this will have no effect at this time, b/c we throw an exception below, but if we would change the flow for some reason to tolerate bad properties, this would make sure we don't add unsupported ones.